onnx/defs/nn/defs.cc

```diff
@@ -822,24 +836,22 @@ void convTransposeShapeInference(InferenceContext& ctx) {
 
       final_output_shape->add_dim()->set_dim_value(output_shape[i]);
     }
-    return; // assume no need to proceed further when the output shape is given.
+    return;
   }
-
-  int kernel_shape_size = static_cast<int>(kernel_shape.size());
-  for (int i = 0; i < kernel_shape_size; ++i) {
-    auto newdim = final_output_shape->add_dim();
-    if (!input_shape.dim(2 + i).has_dim_value()) {
-      continue;
+  else
+  {
+    size_of_output = input_shape.dim_size() - 2;
+    for (int i = 0; i < size_of_output; ++i)
+    {
+      int64_t output_shape_dim =
+          strides[i] * (input_shape.dim(i + 2).dim_value() - 1) +
```

Comment:
The input_shape may have a symbolic dim and not a concrete dim_value. There needs to be a check if input_shape.dim(i+2).has_dim_value() before using that dim_value().